src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -188,10 +198,32 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
 
   List<SourceRecord> sourceRecords(ConsumerRecords<byte[], byte[]> pollResult) {
     List<SourceRecord> sourceRecords = new ArrayList<>(pollResult.count());
-    pollResult.forEach(sourceRecord -> sourceRecords.add(toSourceRecord(sourceRecord)));
+    pollResult.forEach(
+        consumerRecord -> {
+          if (replayPolicy == ReplayPolicy.FILTER && !isSkippedRecord(consumerRecord)) {
+            sourceRecords.add(toSourceRecord(consumerRecord));
+          }
+        });
     return sourceRecords;
   }
 
+  private boolean isSkippedRecord(ConsumerRecord<byte[], byte[]> consumerRecord) {
+    if (replayPolicy == ReplayPolicy.FILTER) {
+      TopicPartition topicPartition =
+          new TopicPartition(consumerRecord.topic(), consumerRecord.partition());
+      long sourceOffset = consumerRecord.offset();
+      Long latestOffset = latestOffsetMap.get(topicPartition);
+      // Skip any record that has already been handled by this task
+      if (latestOffset != null && sourceOffset <= (latestOffset - replayWindowRecords)) {
+        logger.info("Skipping record with offset={}. Latest recorded offset={}", sourceOffset, latestOffset);
```

Comment:
The more I think about this, I don't think it makes sense since we might be skipping hundred of millions of records. Ideally should be a metric emitted. I will remove this.

Reply:
Yeah, that's why I didn't do it in the first place. Would need to do something like only log the first case.